hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/Service.java

```diff
@@ -32,31 +30,4 @@ public interface Service {
      */
     @NonNull
     String getServiceName();
```

Comment:
I think we should have `createHandler` which each service implements that creates service specific handler and pass in needed info to the `ServiceHandlerImpl` constructor.

Reply:
The whole setup part is not clear to me at this point to be honest. There are two questions, I have at the moment.

1. Do we need a generic `createHandler()`?
I do not think that it is needed anywhere. If you need such methods in the `Service` implementations, you will probably need a specific method with a concrete return type, e.g. `CryptoService.createCryptoTransferHandler()`.
2. Do we need such methods in the spi-modules?
My current expectation is that only `hedera-app` will create concrete handlers while services usually get it implicitly via a call-context, but maybe I am wrong. If only `hedera-app` needs it, do we want to define it in the spi-modules?

My suggestion is to leave it for now and revisit the setup part later. Once @tinker-michaelj has a good idea, how we want to DI, we should discuss it.